### DISCUSSION TITLE
AttributeError: 'EMR' object has no attribute 'put_auto_termination_policy'

### DISCUSSION THREAD

**Discussion by pankajmakwana6** [apache/airflow#23573] [General]
### Apache Airflow version

2.0.0

### What happened

not able to update AutoTermination policy of EMR cluster through AWS managed Airflow

  response = client.put_auto_termination_policy(
        ClusterId=job_flow_id,
        AutoTerminationPolicy={
            'IdleTimeout': 5
        }
    )

tried with different version of EMR clusters like 5.30.0 , 6.3.0 etc .

 File "/usr/local/airflow/dags/success_file_test.py", line 37, in auto_term_policy
    response = client.put_auto_termination_policy(
  File "/usr/local/lib/python3.7/site-packages/botocore/client.py", line 616, in __getattr__
    self.__class__.__name__, item)
AttributeError: 'EMR' object has no attribute 'put_auto_termination_policy'
[2022-04-29 05:23:58,727] {{taskinstance.py:1532}} INFO - Marking task as FAILED. dag_id=success-file-test, task_id=run_jobs, execution_date=20220429T052346, start_date=20220429T052351, end_date=20220429T052358

### Operating System

ubuntu

### Deployment

MWAA

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by tirkarthi** [comment]
The `PutAutoTerminationPolicy` was introduced in below commit and seems to be available from botocore version 1.21.31 and above. Can you please add your botocore version?

https://github.com/boto/botocore/commit/fee8473a319dd02df916e9372245ea7699682d9c#diff-9860af22ee80cdd56a1bf858e62f6541924bfd9dd044d85957139322c21a7effR514-R523

```json
    "PutAutoTerminationPolicy":{
      "name":"PutAutoTerminationPolicy",
      "http":{
        "method":"POST",
        "requestUri":"/"
      },
      "input":{"shape":"PutAutoTerminationPolicyInput"},
      "output":{"shape":"PutAutoTerminationPolicyOutput"},
      "documentation":"<p>Creates or updates an auto-termination policy for an Amazon EMR cluster. An auto-termination policy defines the amount of idle time in seconds after which a cluster automatically terminates. For alternative cluster termination options, see <a href=\"https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-plan-termination.html\">Control cluster termination</a>.</p>"
    }
```

**Reply by ss68974** [reply]
the default botocore version for this packge needs to be upgraded

**Comment by potiuk** [comment]
Converted to discussion until we have missing information from the user.
